primefaces/src/main/java/org/primefaces/util/SortTableComparator.java

```diff
@@ -0,0 +1,162 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.primefaces.util;
+
+import java.text.Collator;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Locale;
+import java.util.concurrent.atomic.AtomicInteger;
+import javax.el.ValueExpression;
+import javax.faces.FacesException;
+import javax.faces.context.FacesContext;
+
+import org.primefaces.component.api.UITable;
+import org.primefaces.context.PrimeApplicationContext;
+import org.primefaces.model.SortMeta;
+
+public class SortTableComparator implements Comparator<Object> {
+
+    private final FacesContext context;
+    private final Collection<SortMeta> sortBy;
+    private final UITable<?> table;
+    private final Locale locale;
+    private final String var;
+    private final Collator collator;
+    private final BeanPropertyMapper mapper;
+    private final boolean sortByVEBased;
+
+    public SortTableComparator(FacesContext context, UITable<?> table, BeanPropertyMapper mapper, boolean sortByVEBased) {
+        this.context = context;
+        this.table = table;
+        this.sortBy = table.getActiveSortMeta().values();
+        this.var = table.getVar();
+        this.locale = table.resolveDataLocale(context);
+        this.collator = Collator.getInstance(locale);
+        this.mapper = mapper;
+        this.sortByVEBased = sortByVEBased;
+    }
+
+    public static Comparator<Object> sortByVEBased(FacesContext context, UITable<?> table) {
+        return new SortTableComparator(context, table, sortByVEMapper(), true);
+    }
+
+    public static Comparator<Object> fieldBased(FacesContext context, UITable<?> table) {
+        return new SortTableComparator(context, table, fieldMapper(), false);
+    }
+
+    @Override
+    public int compare(Object o1, Object o2) {
+        AtomicInteger result = new AtomicInteger(0);
+        for (SortMeta sortMeta : sortBy) {
+            if (sortByVEBased) {
```

Comment:
@tandraschko we can do better, and I'll do that in another PR. But if it's not dynamic there is no point calling invokeOnColumn, also don't think the coment is really necessary since you can use p:columns with columngroup